File: 20180622_SDK/SDK/Sample_M50/VB.NET_SBXBPCSample/frmPrtCode.vb
Option Strict Off
Option Explicit On
Friend Class frmSerialNo
	Inherits System.Windows.Forms.Form
	Dim mMachineNumber As Integer
	
	Private Sub cmdGetBackupNumber_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdGetBackupNumber.Click
		Dim vBackupNumber As Integer
		Dim vErrorCode As Integer
		
		lblMessage.Text = "Waiting..."
		txtBackupNo.Text = ""
		System.Windows.Forms.Application.DoEvents()
		
		If sbxpc.SBXPCDLL.EnableDevice(mMachineNumber, False) = False Then
			lblMessage.Text = gstrNoDevice
			Exit Sub
		End If
		
		vBackupNumber = sbxpc.SBXPCDLL.GetBackupNumber(mMachineNumber)
		If vBackupNumber <> 0 Then
			txtBackupNo.Text = CStr(vBackupNumber)
			lblMessage.Text = "Success"
		Else
			sbxpc.SBXPCDLL.GetLastError(frmMain.gMachineNumber, vErrorCode)
			lblMessage.Text = ErrorPrint(vErrorCode)
		End If
		
		sbxpc.SBXPCDLL.EnableDevice(mMachineNumber, True)
	End Sub
	
	Private Sub cmdGetProductCode_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdGetProductCode.Click
        Dim vProductCode As String = ""
		Dim vRet As Integer
		Dim vErrorCode As Integer
		
		lblMessage.Text = "Waiting..."
		txtProductCode.Text = ""
		System.Windows.Forms.Application.DoEvents()
		
		vRet = sbxpc.SBXPCDLL.EnableDevice(mMachineNumber, False)
		If vRet = False Then
			lblMessage.Text = gstrNoDevice
			Exit Sub
		End If
		
        vRet = sbxpc.SBXPCDLL.GetProductCode(mMachineNumber, vProductCode)
		If vRet = True Then
			txtProductCode.Text = vProductCode
			lblMessage.Text = "Success"
		Else
			sbxpc.SBXPCDLL.GetLastError(frmMain.gMachineNumber, vErrorCode)
			lblMessage.Text = ErrorPrint(vErrorCode)
		End If
		
		sbxpc.SBXPCDLL.EnableDevice(mMachineNumber, True)
	End Sub
	
	Private Sub cmdGetSerialNumber_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdGetSerialNumber.Click
        Dim vSerialNumber As String = ""
		Dim vRet As Integer
		Dim vErrorCode As Integer
		
		lblMessage.Text = "Waiting..."
		txtSerialNo.Text = ""
		System.Windows.Forms.Application.DoEvents()
		
		vRet = sbxpc.SBXPCDLL.EnableDevice(mMachineNumber, False)
		If vRet = False Then
			lblMessage.Text = gstrNoDevice
			Exit Sub
		End If
		
        vRet = sbxpc.SBXPCDLL.GetSerialNumber(mMachineNumber, vSerialNumber)
		If vRet = True Then
			txtSerialNo.Text = vSerialNumber
			lblMessage.Text = "Success"
		Else
			sbxpc.SBXPCDLL.GetLastError(frmMain.gMachineNumber, vErrorCode)
			lblMessage.Text = ErrorPrint(vErrorCode)
		End If
		
		sbxpc.SBXPCDLL.EnableDevice(mMachineNumber, True)
	End Sub
	
	Private Sub cmdExit_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdExit.Click
		Me.Close()
    End Sub

    Private Sub frmSerialNo_FormClosed(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
        frmMain.Visible = True
    End Sub
	
	Private Sub frmSerialNo_Load(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Load
		mMachineNumber = frmMain.gMachineNumber
	End Sub
	
    '	Private Sub frmSerialNo_FormClosed(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
    '		Me.Close()
    '		frmMain.Visible = True
    '	End Sub
End Class

File: 20180622_SDK/SDK/Sample_M50/C#_SBXPCDLL_Sample/SBXPCDLLSampleCSharp/frmPrtCode.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace SBXPCDLLSampleCSharp
{
    public partial class frmPrtCode : Form
    {
        public frmPrtCode()
        {
            InitializeComponent();
        }

        private void cmdGetSerialNumber_Click(object sender, EventArgs e)
        {
            string vSerialNumber = "";
            Boolean vRet;
            int vErrorCode = 0;

            txtSerialNo.Text = "";
            lblMessage.Text = "Waiting...";
            Application.DoEvents();

            vRet = sbxpc.SBXPCDLL.EnableDevice(Program.gMachineNumber, 0); // 0 : false
            if (!vRet)
            {
                lblMessage.Text = util.gstrNoDevice;
                return;
            }

            vRet = sbxpc.SBXPCDLL.GetSerialNumber(Program.gMachineNumber, out vSerialNumber);
            if (vRet)
            {
                txtSerialNo.Text = vSerialNumber;
                lblMessage.Text = "Success";
            }
            else
            {
                sbxpc.SBXPCDLL.GetLastError(Program.gMachineNumber, out vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
            }

            sbxpc.SBXPCDLL.EnableDevice(Program.gMachineNumber, 1); // 1 : true
        }

        private void cmdGetBackupNumber_Click(object sender, EventArgs e)
        {
            int vBackupNumber;
            int vErrorCode = 0;
            Boolean vRet;

            lblMessage.Text = "Waiting...";
            txtBackupNo.Text = "";
            Application.DoEvents();

            vRet = sbxpc.SBXPCDLL.EnableDevice(Program.gMachineNumber, 0); // 0 : false
            if (!vRet)
            {
                lblMessage.Text = util.gstrNoDevice;
                return;
            }

            vBackupNumber = sbxpc.SBXPCDLL.GetBackupNumber(Program.gMachineNumber);
            if (vBackupNumber != 0)
            {
                txtBackupNo.Text = Convert.ToString(vBackupNumber);
                lblMessage.Text = "Success";
            }
            else
            {
                sbxpc.SBXPCDLL.GetLastError(Program.gMachineNumber, out vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
            }

            sbxpc.SBXPCDLL.EnableDevice(Program.gMachineNumber, 1); // 1 : true
        }

        private void cmdGetProductCode_Click(object sender, EventArgs e)
        {
            string vProductCode = "";
            Boolean vRet;
            int vErrorCode = 0;

            txtProductCode.Text = "";
            lblMessage.Text = "Waiting...";
            Application.DoEvents();

            vRet = sbxpc.SBXPCDLL.EnableDevice(Program.gMachineNumber, 0); // 0 : false
            if (!vRet)
            {
                lblMessage.Text = util.gstrNoDevice;
                return;
            }

            vRet = sbxpc.SBXPCDLL.GetProductCode(Program.gMachineNumber, out vProductCode);
            if (vRet)
            {
                txtProductCode.Text = vProductCode;
                lblMessage.Text = "Success";
            }
            else
            {
                sbxpc.SBXPCDLL.GetLastError(Program.gMachineNumber, out vErrorCode);
                lblMessage.Text = util.ErrorPrint(vErrorCode);
            }

            sbxpc.SBXPCDLL.EnableDevice(Program.gMachineNumber, 1); // 1 : true
        }

        private void cmdExit_Click(object sender, EventArgs e)
        {
            Close();
            Application.OpenForms["frmMain"].Visible = true;
        }

        private void frmPrtCode_FormClosed(object sender, FormClosedEventArgs e)
        {
            Application.OpenForms["frmMain"].Visible = true;
        }

        private void frmPrtCode_Load(object sender, EventArgs e)
        {
        }
    }
}
